src/commands/react_native/xcode.rs

```diff
@@ -371,21 +365,43 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
                 chunk_upload_options: chunk_upload_options.as_ref(),
             })?;
         } else {
-            let dist = dist_from_env.unwrap_or_else(|_| plist.build().to_string());
-            let release_name = release_from_env.unwrap_or(format!(
-                "{}@{}+{}",
-                plist.bundle_id(),
-                plist.version(),
-                dist
-            ));
+            let mut dist: Option<String> = None;
+            let mut release_name: Option<String> = None;
+
+            if dist_from_env.is_err() && release_from_env.is_err() {
+              info!("Parsing Info.plist");
+              let plist = match InfoPlist::discover_from_env()? {
+                  Some(plist) => plist,
+                  None => bail!("Could not find info.plist"),
+              };
+              info!("Parse result from Info.plist: {:?}", &plist);
+
+              let dist_string = plist.build().to_string();
+              dist = Some(dist_string.clone());
+              release_name = Some(format!(
+                  "{}@{}+{}",
+                  plist.bundle_id(),
+                  plist.version(),
+                  dist_string
+              ));
+            }
+
+            if let Ok(_dist) = dist_from_env {
+              info!("Using dist from `SENTRY_DIST` env");
+              dist = Some(_dist);
+            }
+            if let Ok(_release) = release_from_env {
+              info!("Using release from `SENTRY_RELEASE` env");
+              release_name = Some(_release);
+            }
```

Comment:
I find this logic somewhat difficult to follow – I would prefer, if possible, to set the `dist` and `release_name` in one place, ideally keeping them immutable. Perhaps this could be achieved with a `match` statement?

Reply:
Also, please check the indentation (the linter/formatter might be checking it already, but it appears the code is inconsistently combining both 2- and 4-space indentations)

Reply:
Thank you, yes I agree, I've used match, and now it's much simpler!